x/mint/keeper/genesis_test.go

```diff
@@ -4,26 +4,183 @@ import (
+			},
+			{
+				Address: "osmo1gw445ta0aqn26suz2rg3tkqfpxnq2hs224d7gq",
+				Weight:  sdk.NewDecWithPrec(4, 1),
+			},
+		},
+		2), // minting reward distribution start epoch
+	3) // halven started epoch
+
+func TestMintGenesisTestSuite(t *testing.T) {
+	suite.Run(t, new(KeeperTestSuite))
+}
+
+// TestMintInitGenesis tests that genesis is initialized correctly
+// with different parameters and state.
+func (suite *KeeperTestSuite) TestMintInitGenesis() {
+	testCases := map[string]struct {
+		mintGenesis                     *types.GenesisState
+		mintDenom                       string
+		ctxHeight                       int64
+		isDeveloperModuleAccountCreated bool
+
+		expectPanic             bool
+		expectedEpochProvisions sdk.Dec
+		// Deltas represent by how much a certain paramets
+		// has changed after calling InitGenesis()
+		expectedSupplyOffsetDelta           sdk.Int
+		expectedSupplyWithOffsetDelta       sdk.Int
+		expectedDeveloperVestingAmountDelta sdk.Int
+		expectedHalvenStartedEpoch          int64
+	}{
+		"default genesis - developer module account is not created prior to InitGenesis() - created during the call": {
+			mintGenesis: types.DefaultGenesisState(),
+			mintDenom:   sdk.DefaultBondDenom,
+
+			expectedEpochProvisions:             types.DefaultGenesisState().Params.GenesisEpochProvisions,
+			expectedSupplyOffsetDelta:           sdk.NewInt(keeper.DeveloperVestingAmount).Neg(),
+			expectedSupplyWithOffsetDelta:       sdk.ZeroInt(),
+			expectedDeveloperVestingAmountDelta: sdk.NewInt(keeper.DeveloperVestingAmount),
+		},
+		"default genesis - developer module account is created prior to InitGenesis() - not created during the call": {
+			mintGenesis:                     types.DefaultGenesisState(),
+			mintDenom:                       sdk.DefaultBondDenom,
+			isDeveloperModuleAccountCreated: true,
+
+			expectedEpochProvisions:             types.DefaultGenesisState().Params.GenesisEpochProvisions,
+			expectedSupplyOffsetDelta:           sdk.ZeroInt(),
+			expectedSupplyWithOffsetDelta:       sdk.ZeroInt(),
+			expectedDeveloperVestingAmountDelta: sdk.ZeroInt(),
+		},
+		"custom genesis": {
+			mintGenesis: customGenesis,
+			mintDenom:   "uosmo",
+
+			expectedEpochProvisions:             sdk.NewDec(200),
+			expectedSupplyOffsetDelta:           sdk.NewInt(keeper.DeveloperVestingAmount).Neg(),
+			expectedSupplyWithOffsetDelta:       sdk.ZeroInt(),
+			expectedDeveloperVestingAmountDelta: sdk.NewInt(keeper.DeveloperVestingAmount),
+			expectedHalvenStartedEpoch:          3,
+		},
+		"nil genesis state - panic": {
+			mintDenom:   sdk.DefaultBondDenom,
+			expectPanic: true,
+		},
+	}
+
+	for name, tc := range testCases {
+		suite.Run(name, func() {
+			// Setup.
+			suite.setupDeveloperVestingModuleAccountTest(tc.ctxHeight, tc.isDeveloperModuleAccountCreated)
+			ctx := suite.Ctx
+			accountKeeper := suite.App.AccountKeeper
+			bankKeeper := suite.App.BankKeeper
+			mintKeeper := suite.App.MintKeeper
```

Comment:
Something we could adapt our test standards is aliasing relevant keeper in the module's suite, if we find this helpful

Reply:
I like the idea. I will document this in an issue to further think about

Reply:
https://github.com/osmosis-labs/osmosis/issues/1947